
![TOC](https://user-images.githubusercontent.com/95513476/151341165-cf155948-dffa-401a-b987-0f969181f8d3.jpg)
Understanding the mutational genotype-disease phenotype relationship is a fundamental step for enabling precision medicine. It remains a challenging task to assess the molecular principle of the genotype- phenotype relationship. By focusing on Hypophosphatasia, a rare inherited disorder, as an example, we performed a comprehensive analysis of single-amino-acid mutations in the encoded protein of Tissue Nonspecific Alkaline Phosphatase associated in terms of their sequence, structure, and dynamic properties. We further developed a machine learning-based method to classify different disease phenotypes, and the interpretability of the classification model was addressed by the structural-functional analysis of network topologically important mutations. Our results highlight the allosteric propensity of severe mutations, and show the potential allosteric principle of genotype-phenotype relationships.  

## Dissecting Mutational Allosteric Effects in Alkaline Phosphatases Associated with Different Hypophosphatasia Phenotypes: An Integrative Computational Investigation
Fei Xiao[^1], Ziyun Zhou[^1], Xingyu Song[^2], Mi Gan[^1], Jie Long[^1], Gennady M Verkhivker[^3][^4], Guang Hu[^1]*
[^1]: My reference.Center for Systems Biology, Department of Bioinformatics, School of Biology and Basic Medical Sciences, Soochow University, Suzhou 215123, China
[^2]: Department of Chemistry, Institutes of Biomedical Sciences, Fudan University, Shanghai 200438, China
[^3]: Department of Computational and Data Sciences, Chapman University, One University Drive, Orange, California 92866, USA
[^4]: Department of Biomedical and Pharmaceutical Sciences, Chapman University Pharmacy School 9401 Jeronimo Rd, Irvine, CA 92618, USA
  
#### Data and Tools
*Mutation data is download from four public database:*
- ALPL gene mutation database: https://alplmutationdatabase.jku.at
- Human Gene Mutation database: http://www.hgmd.cf.ac.uk/ac/index.php
- Locus Specific Mutation database: http://www.hgmd.cf.ac.uk/docs/oth_mut.html
- PubMed: https://pubmed.ncbi.nlm.nih.gov

*All utilized computational tools include servers and programs are freely available for academic users:*
- Consurf server: https://consurf.tau.ac.il
- MODELLER 9.19: https://salilab.org/modeller
- FoldX :http://foldxsuite.crg.eu/ 
- PSAIA 1.0: http://complex.zesoi.fer.hr/tools/PSAIA.html 
- ProDy 1.10.10: http://prody.csb.pitt.edu
- R package NACEN 0.1.0: http://sysbio.suda.edu.cn/NACEN/index.html
- R package caret 6.0.86: https://CRAN.R-project.org/package=caret
- R package random forest 4.6.14: https://www.stat.berkeley.edu/~breiman/RandomForests
- R package pROC 1.17.0.1: https://cran.r-project.org/web/packages/pROC/
- Gromacs 5.1.4: https://manual.gromacs.org/documentation/5.1.4/index.html  
- MD-TASK: https://github.com/RUBi-ZA/MD-TASK.git  

*For requirement of more data and code, please contact zhouziyun1900@hotmail.com*  
